src/dune_rules/lib_flags.ml

```diff
@@ -92,75 +92,99 @@ module L = struct
        |> List.rev)
   ;;
 
+  let to_flags dirs ~dirs_hidden =
+    Command.Args.S
+      ((Path.Set.fold dirs ~init:[] ~f:(fun dir acc ->
+        if Path.Set.mem dirs_hidden dir then
```

Comment:
This seem too complex. Instead of using a set, one should use a map now, which maps libraries to the boolean, and here you should just fold on the map bindings.